services/api-server/src/simcore_service_api_server/api/dependencies/authentication.py

```diff
@@ -2,16 +2,22 @@
 
 from fastapi import Depends, HTTPException, Security, status
 from fastapi.security import HTTPBasic, HTTPBasicCredentials
-from pydantic.types import PositiveInt
+from pydantic import BaseModel, PositiveInt
 
-from ...db.repositories.api_keys import ApiKeysRepository
+from ...db.repositories.api_keys import ApiKeysRepository, UserAndProductTuple
 from ...db.repositories.users import UsersRepository
 from .database import get_repository
 
 # SEE https://swagger.io/docs/specification/authentication/basic-authentication/
 basic_scheme = HTTPBasic()
 
 
+class Identity(BaseModel):
+    user_id: PositiveInt
+    product_name: str
+    email: str
```

Comment:
pydantic has a validator for email, something like EmailStr and I see we also have LowerCaseEmailStr... maybe worth checking?

Reply:
Good point. I will make that change in a second follow-up PR about the products in the api server